src/Microsoft.Health.Fhir.Api/Controllers/ExportController.cs

```diff
@@ -0,0 +1,131 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Net;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    public class ExportController : Controller
+    {
+        /*
+         * We are currently hardcoding the routing attribute to be specific to Export and
+         * get forwarded to this controller. As we add more operations we would like to resolve
+         * the routes in a more dynamic manner. One way would be to use a regex route constraint
+         * - eg: "{operation:regex(^\\$([[a-zA-Z]]+))}" - and use the appropriate operation handler.
+         * Another way would be to use the capability statement to dynamically determine what operations
+         * are supported.
+         * It would be easier to determine what pattern to follow once we have built support for a couple
+         * of operations. Then we can refactor this controller accordingly.
+         */
+
+        private readonly ILogger<ExportController> _logger;
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly ExportConfiguration _exportConfig;
+
+        public ExportController(
+            ILogger<ExportController> logger,
+            IFhirRequestContextAccessor fhirRequestContextAccessor,
+            IOptions<ExportConfiguration> exportConfig)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(exportConfig, nameof(exportConfig));
```

Comment:
validate against exportConfig?.Value to make sure exportConfig is not null and its value is also not null.